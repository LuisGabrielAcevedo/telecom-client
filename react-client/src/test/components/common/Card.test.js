import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Card from "../../../components/common/Card";
import formattedWeatherResp from "../../mocks/formattedWeatherResp.mock";
import { format } from "date-fns";
import { es } from "date-fns/locale";

describe("Card Component tests", () => {
  const wrapper = shallow(<Card {...formattedWeatherResp} />);
  const formatDate = (_date) => {
    return format(new Date(_date.split(" ")[0]), "dd EEEE", { locale: es });
  };

  it("the component should be initialized", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("the component should render the title entered by props", () => {
    const title = wrapper.find(".card__title").text();
    expect(title).toBe(formattedWeatherResp.title);
  });

  it("the component should render the temperature entered by props in C °", () => {
    const formattedTemp =
      (formattedWeatherResp.temp - 273).toFixed(2).toString() + " °";
    const temp = wrapper.find(".card__temp").text();
    expect(temp).toBe(formattedTemp);
  });

  it("the component should render 5 items", () => {
    expect(wrapper.find(".card__item")).toHaveLength(5);
  });

  it("the component should render 5 items", () => {
    const date = wrapper.find(".card__date").first().text();
    const formattedDate = formatDate(formattedWeatherResp.nextDays[0].dt_txt);
    expect(date).toBe(formattedDate);
  });
});
