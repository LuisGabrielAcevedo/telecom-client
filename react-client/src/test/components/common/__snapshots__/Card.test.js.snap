// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component tests the component should be initialized 1`] = `
<div
  className="card"
>
  <div
    className="card__row"
  >
    <span
      className="card__title"
    >
      Buenos Aires F.D.
    </span>
    <span
      className="card__temp"
    >
      13.57
       °
    </span>
  </div>
  <div
    className="card__extra"
  >
    <div
      className="card__column"
    >
      <span>
        Viento
      </span>
      <span>
        1
         Km/h
      </span>
    </div>
    <div
      className="card__column"
    >
      <span>
        Humedad
      </span>
      <span>
        100
         %
      </span>
    </div>
  </div>
  <div
    className="card__item"
    key="0"
  >
    <span
      className="card__date"
    >
      21 martes
    </span>
    <span>
      14.12
    </span>
  </div>
  <div
    className="card__item"
    key="1"
  >
    <span
      className="card__date"
    >
      22 miércoles
    </span>
    <span>
      14.60
    </span>
  </div>
  <div
    className="card__item"
    key="2"
  >
    <span
      className="card__date"
    >
      23 jueves
    </span>
    <span>
      12.48
    </span>
  </div>
  <div
    className="card__item"
    key="3"
  >
    <span
      className="card__date"
    >
      24 viernes
    </span>
    <span>
      9.28
    </span>
  </div>
  <div
    className="card__item"
    key="4"
  >
    <span
      className="card__date"
    >
      25 sábado
    </span>
    <span>
      11.38
    </span>
  </div>
</div>
`;
